File: GianniPasquinelli/phyton/leccion14/mundo_pc/Orden.py
class Orden:
  
  contador_ordenes = 0
  
  def __init__(self, computadoras):
    Orden.contador_ordenes += 1
    self.__id_orden = Orden.contador_ordenes
    self.computadoras = computadoras
    
  def agregar_computadora(self, computadora):
    self.computadoras.append(computadora)
    
  def mostrar_orden(self):
    computadoras_str = ''
    for computadora in self.computadoras:
      computadoras_str += computadora.__str__() + '\n'
    return f'Orden: {self.__id_orden}\nComputadoras:\n{computadoras_str}'
  
  def __str__(self):
    return self.mostrar_orden()

File: Santiago-Villarreal/Programacion/Java/TercerSemestre/Leccion 1/MundoPC/src/ar/com/system2023/MundoPC/Orden.java

package ar.com.system2023.MundoPC;

public class Orden {
    private final int idOrden;
    private Computadora computadora[];
    private static int contadorOrdenes;
    private static final int MAX_COMPUTADORAS = 6;
    private int contadorComputadora;
    
    public Orden(){
        this.idOrden = ++Orden.contadorOrdenes;
        this.computadora  = new Computadora[Orden.MAX_COMPUTADORAS];
    }
    
    public void agregarComputadora(Computadora computadora){
        if(this.contadorComputadora < Orden.MAX_COMPUTADORAS){
            this.computadora[this.contadorComputadora++] = computadora;
        }
        else{
            System.out.println("Has superado el límite: " + Orden.MAX_COMPUTADORAS);
        }
    }
    public void mostrarOrden(){
        System.out.println("Orden #: " + this.idOrden);
        System.out.println("Computadoras de la orden #: " + this.idOrden);
        for(int i = 0; i < this.contadorComputadora; i++){
            System.out.println(this.computadora[i]);
        }
    }
}